File: back/pl-atv-5/src/main/java/com/fatec/pl/modelo/Rg.java
package com.fatec.pl.modelo;

import java.util.Date;

import org.springframework.hateoas.RepresentationModel;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
public class Rg extends RepresentationModel<Rg>{

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@Column
	private String valor;
	
	@Column
	private Date dataEmissao;
}

File: front/src/modelo/rg.ts

export default interface RG {
    id?: number,
    valor: string,
    dataEmissao: Date
}

// export default class RG {
//     private valor: string
//     private dataEmissao: Date
//     constructor(valor: string, dataEmissao: Date) {
//         this.valor = valor
//         this.dataEmissao = dataEmissao
//     }
//     public get getValor(): string {
//         return this.valor
//     }
//     public get getDataEmissao(): Date {
//         return this.dataEmissao
//     }
// }